unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/States/EditorMode/BuildEditorMode.cs

```diff
@@ -0,0 +1,492 @@
+using Builder.Gizmos;
+using DCL;
+using DCL.Configuration;
+using DCL.Controllers;
+using DCL.Helpers;
+using DCL.Models;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class BuildEditorMode : BuildModeState
+{
+    [Header("Editor Design")]
+    public float distanceEagleCamera = 20f;
+
+    [Header("Scenes References")]
+    public FreeCameraMovement freeCameraController;
+    public DCLBuilderGizmoManager gizmoManager;
+    public VoxelController voxelController;
+    public BuilderInputWrapper builderInputWrapper;
+    public OutlinerController outlinerController;
+    public BuildModeController buildModeController;
+    public CameraController cameraController;
+    public Transform lookAtT;
+    public MouseCatcher mouseCatcher;
+
+    [Header("InputActions")]
+
+    [SerializeField] internal InputAction_Trigger focusOnSelectedEntitiesInputAction;
+
+    [SerializeField] internal InputAction_Hold squareMultiSelectionInputAction;
+
+
+    ParcelScene sceneToEdit;
+
+    public LayerMask groundLayer;
+
+    bool isPlacingNewObject = false, mousePressed = false, isMakingSquareMultiSelection = false,isTypeOfBoundSelectionSelected = false, isVoxelBoundMultiSelection = false,squareMultiSelectionButtonPressed = false;
```

Comment:
We should avoid too long lines of code like this.
```suggestion
    bool isPlacingNewObject = false,
         mousePressed = false,
         isMakingSquareMultiSelection = false,
         isTypeOfBoundSelectionSelected = false,
         isVoxelBoundMultiSelection = false,
         squareMultiSelectionButtonPressed = false;
```